framework/db/src/main/java/com/cloud/utils/crypt/EncryptionSecretKeyChanger.java

```diff
@@ -22,149 +22,332 @@
+        Option oldDBKey = Option.builder("d").longOpt(OLD_DB_KEY_OPTION).argName(OLD_DB_KEY_OPTION).required(true).hasArg().desc("(required) Current DB Secret Key").build();
+        Option newMSKey = Option.builder("n").longOpt(NEW_MS_KEY_OPTION).argName(NEW_MS_KEY_OPTION).required(false).hasArg().desc("New Mgmt Secret Key").build();
+        Option newDBKey = Option.builder("e").longOpt(NEW_DB_KEY_OPTION).argName(NEW_DB_KEY_OPTION).required(false).hasArg().desc("New DB Secret Key").build();
+        Option encryptorVersion = Option.builder("v").longOpt(ENCRYPTOR_VERSION_OPTION).argName(ENCRYPTOR_VERSION_OPTION).required(false).hasArg().desc("New DB Encryptor Version. Options are V1, V2.").build();
+
+        Option loadNewMsKeyFromEnv = Option.builder("l").longOpt(LOAD_NEW_MS_KEY_FROM_ENV_FLAG).desc("Load new management key from environment variable " + ENV_NEW_MANAGEMENT_KEY).build();
+        Option forceDatabaseMigration = Option.builder("f").longOpt(FORCE_DATABASE_MIGRATION_FLAG).desc("Force database migration even if DB Secret key is not changed").build();
+        Option skipDatabaseMigration = Option.builder("s").longOpt(SKIP_DATABASE_MIGRATION_FLAG).desc("Skip database migration even if DB Secret key is changed").build();
+        Option help = Option.builder("h").longOpt(HELP_FLAG).desc("Show help message").build();
+
+        options.addOption(oldMSKey);
+        options.addOption(oldDBKey);
+        options.addOption(newMSKey);
+        options.addOption(newDBKey);
+        options.addOption(encryptorVersion);
+        options.addOption(loadNewMsKeyFromEnv);
+        options.addOption(forceDatabaseMigration);
+        options.addOption(skipDatabaseMigration);
+        options.addOption(help);
+
+        return options;
+    }
+
+    private static HelpFormatter initializeHelper() {
+        HelpFormatter helper = new HelpFormatter();
+
+        helper.setOptionComparator((o1, o2) -> {
+            if (o1.isRequired() && !o2.isRequired()) {
+                return -1;
             }
-            // Old DB Key
-            if (arg.equals("-d")) {
-                oldDBKey = iter.next();
+            if (!o1.isRequired() && o2.isRequired()) {
+                return 1;
             }
-            // New MS Key
-            if (arg.equals("-n")) {
-                newMSKey = iter.next();
+            if (o1.hasArg() && !o2.hasArg()) {
+                return -1;
             }
-            // New DB Key
-            if (arg.equals("-e")) {
-                newDBKey = iter.next();
+            if (!o1.hasArg() && o2.hasArg()) {
+                return 1;
             }
-        }
+            return o1.getOpt().compareTo(o2.getOpt());
+        });
+
+        return helper;
+    }
+
+    private static boolean validateParameters(String oldMSKey, String oldDBKey, String newMSKey, String newDBKey,
```

Comment:
this method is doing a lot, it could be dissected into smaller methods.

Reply:
@DaanHoogland 
thanks for review.
`validateParameters` looks ok to me. 

`migrateEverything` is too large but I'd like to keep it at this stage (this PR has been tested by QA).
Actually most lines in the method are validation. The migrations have been already extracted to some methods.